### PR TITLE
topsql: fix race in test

### LINKED ISSUE #31713: DATA RACE in the ```stmtstats.Test_aggregator_disable_aggregate```
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

make race 

```
==================
WARNING: DATA RACE
Read at 0x00c00019c270 by goroutine 46:
  reflect.maplen()
      /usr/local/go/src/runtime/map.go:1360 +0x0
  reflect.Value.Len()
      /usr/local/go/src/reflect/value.go:1496 +0x1f4
  github.com/stretchr/testify/assert.getLen()
      /home/prow/go/pkg/mod/github.com/stretchr/testify@v1.7.0/assert/assertions.go:627 +0x207
  github.com/stretchr/testify/assert.Len()
      /home/prow/go/pkg/mod/github.com/stretchr/testify@v1.7.0/assert/assertions.go:638 +0xb1
  github.com/pingcap/tidb/util/topsql/stmtstats.Test_aggregator_disable_aggregate()
      /go/tidb/util/topsql/stmtstats/aggregator_test.go:112 +0x528
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /usr/local/go/src/testing/testing.go:1306 +0x47
Previous write at 0x00c00019c270 by goroutine 47:
  runtime.mapassign()
      /usr/local/go/src/runtime/map.go:571 +0x0
  github.com/pingcap/tidb/util/topsql/stmtstats.StatementStatsMap.Merge()
      /go/tidb/util/topsql/stmtstats/stmtstats.go:159 +0x1d6
  github.com/pingcap/tidb/util/topsql/stmtstats.Test_aggregator_disable_aggregate.func1()
      /go/tidb/util/topsql/stmtstats/aggregator_test.go:90 +0x3a
  github.com/pingcap/tidb/util/topsql/stmtstats.(*mockCollector).CollectStmtStatsMap()
      /go/tidb/util/topsql/stmtstats/aggregator_test.go:128 +0x45
  github.com/pingcap/tidb/util/topsql/stmtstats.(*aggregator).aggregate.func2()
      /go/tidb/util/topsql/stmtstats/aggregator.go:83 +0x56
  sync.(*Map).Range()
      /usr/local/go/src/sync/map.go:346 +0x205
  github.com/pingcap/tidb/util/topsql/stmtstats.(*aggregator).aggregate()
      /go/tidb/util/topsql/stmtstats/aggregator.go:82 +0xba
  github.com/pingcap/tidb/util/topsql/stmtstats.(*aggregator).run()
      /go/tidb/util/topsql/stmtstats/aggregator.go:63 +0x315
  github.com/pingcap/tidb/util/topsql/stmtstats.Test_aggregator_disable_aggregate.func2()
      /go/tidb/util/topsql/stmtstats/aggregator_test.go:96 +0x39
Goroutine 46 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1306 +0x726
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1598 +0x99
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1259 +0x22f
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1596 +0x7ca
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1504 +0x9d1
  go.uber.org/goleak.VerifyTestMain()
      /home/prow/go/pkg/mod/go.uber.org/goleak@v1.1.12/testmain.go:53 +0x59
  github.com/pingcap/tidb/util/topsql/stmtstats.TestMain()
      /go/tidb/util/topsql/stmtstats/main_test.go:26 +0x44
  main.main()
      _testmain.go:65 +0x264
Goroutine 47 (running) created at:
  github.com/pingcap/tidb/util/topsql/stmtstats.Test_aggregator_disable_aggregate()
      /go/tidb/util/topsql/stmtstats/aggregator_test.go:95 +0x309
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /usr/local/go/src/testing/testing.go:1306 +0x47
==================
==================
WARNING: DATA RACE
Read at 0x00c0005b6010 by goroutine 46:
  github.com/pingcap/tidb/util/topsql/stmtstats.(*aggregator).close()
      /go/tidb/util/topsql/stmtstats/aggregator.go:120 +0x544
  github.com/pingcap/tidb/util/topsql/stmtstats.Test_aggregator_disable_aggregate()
      /go/tidb/util/topsql/stmtstats/aggregator_test.go:115 +0x52e
      /go/tidb/util/topsql/stmtstats/aggregator_test.go:95 +0x309
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /usr/local/go/src/testing/testing.go:1306 +0x47
==================
==================
WARNING: DATA RACE
Write at 0x00c00059c090 by goroutine 46:
  sync/atomic.CompareAndSwapInt32()
      /usr/local/go/src/runtime/race_amd64.s:334 +0xb
  sync/atomic.CompareAndSwapInt32()
      <autogenerated>:1 +0x1e
  context.(*cancelCtx).cancel()
      /usr/local/go/src/context/context.go:401 +0x84
  context.WithCancel.func1()
      /usr/local/go/src/context/context.go:238 +0x4f
  github.com/pingcap/tidb/util/topsql/stmtstats.(*aggregator).close()
      /go/tidb/util/topsql/stmtstats/aggregator.go:120 +0x552
  github.com/pingcap/tidb/util/topsql/stmtstats.Test_aggregator_disable_aggregate()
      /go/tidb/util/topsql/stmtstats/aggregator_test.go:115 +0x52e
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /usr/local/go/src/testing/testing.go:1306 +0x47
Previous write at 0x00c00059c090 by goroutine 47:
  context.WithCancel()
      /usr/local/go/src/context/context.go:236 +0xa4
  github.com/pingcap/tidb/util/topsql/stmtstats.(*aggregator).run()
      /go/tidb/util/topsql/stmtstats/aggregator.go:50 +0x5c
  github.com/pingcap/tidb/util/topsql/stmtstats.Test_aggregator_disable_aggregate.func2()
      /go/tidb/util/topsql/stmtstats/aggregator_test.go:96 +0x39
Goroutine 46 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1306 +0x726
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1598 +0x99
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1259 +0x22f
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1596 +0x7ca
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1504 +0x9d1
  go.uber.org/goleak.VerifyTestMain()
      /home/prow/go/pkg/mod/go.uber.org/goleak@v1.1.12/testmain.go:53 +0x59
  github.com/pingcap/tidb/util/topsql/stmtstats.TestMain()
      /go/tidb/util/topsql/stmtstats/main_test.go:26 +0x44
  main.main()
      _testmain.go:65 +0x264
Goroutine 47 (running) created at:
  github.com/pingcap/tidb/util/topsql/stmtstats.Test_aggregator_disable_aggregate()
      /go/tidb/util/topsql/stmtstats/aggregator_test.go:95 +0x309
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /usr/local/go/src/testing/testing.go:1306 +0x47
==================
```

### PR DESCRIPTION
**Pull request by mornyx**
Signed-off-by: mornyx <mornyx.z@gmail.com>

Issue Number: close #31713

### Release note

```release-note
None
```

### CODE REVIEW

File: util/topsql/stmtstats/aggregator_test.go [hunk 1]
```diff
@@ -106,10 +109,14 @@ func Test_aggregator_disable_aggregate(t *testing.T) {
 	state.DisableTopSQL()
 	a.register(stats)
 	time.Sleep(1500 * time.Millisecond)
+	mu.Lock()
 	assert.Empty(t, total)
+	mu.Unlock()
 	state.EnableTopSQL()
 	time.Sleep(1500 * time.Millisecond)
+	mu.Lock()
 	assert.Len(t, total, 1)
```
Comment (hawkingrei) [inline util/topsql/stmtstats/aggregator_test.go:118]: use ```require.Len```.

File: util/topsql/stmtstats/aggregator_test.go [hunk 2]
```diff
@@ -106,10 +109,14 @@ func Test_aggregator_disable_aggregate(t *testing.T) {
 	state.DisableTopSQL()
 	a.register(stats)
 	time.Sleep(1500 * time.Millisecond)
+	mu.Lock()
 	assert.Empty(t, total)
```
Comment (hawkingrei) [inline util/topsql/stmtstats/aggregator_test.go:113]: should use ```require.Empty```

File: util/topsql/stmtstats/aggregator_test.go [hunk 3]
```diff
@@ -84,10 +84,13 @@ func Test_aggregator_run_close(t *testing.T) {
 }
 
 func Test_aggregator_disable_aggregate(t *testing.T) {
```
Comment (hawkingrei) [inline util/topsql/stmtstats/aggregator_test.go:86]: use ```TestAggregatorDisableAggregate```.

Comment (mornyx) [reply util/topsql/stmtstats/aggregator_test.go:86]: I'm not sure, this test is for private type, still use big camel case?

**Review by hawkingrei — APPROVED** [review]

**Review by crazycs520 — APPROVED** [review]

**Pull request merged**